app/auth/strategies/discord.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable no-param-reassign */
+const { Strategy } = require('passport-discord');
+const { clientID, clientSecret } = require('../../config').oauth.discord;
+const User = require('../../db/models/user.model');
+const registerUser = require('../../functions/registerUser');
+const generateRandomBytes = require('../../functions/generateRandomBytes');
+
+module.exports = (passport, refresh) => {
```

Comment:
How much of these different strategies can we consolidate into generic functions?

Reply:
I thought that this was a good idea for the sake of modularity. I know that they are going against DRY, but I can't think of a better way to write them right now. Any ideas?

Reply:
Refactored this bit. Ready for another review